pkg/resource/deploy/step_generator.go

```diff
@@ -454,7 +469,8 @@ func (sg *stepGenerator) GenerateDeletes() []Step {
 				logging.V(7).Infof("Planner decided to delete '%v' due to replacement", res.URN)
 				sg.deletes[res.URN] = true
 				dels = append(dels, NewDeleteReplacementStep(sg.plan, res, false))
-			} else if !sg.sames[res.URN] && !sg.updates[res.URN] && !sg.replaces[res.URN] && !sg.reads[res.URN] {
+			} else if _, aliased := sg.aliased[res.URN]; !sg.sames[res.URN] && !sg.updates[res.URN] && !sg.replaces[res.URN] &&
+				!sg.reads[res.URN] && !aliased {
```

Comment:
would love if this was extracted into something nice and easily named like "shouldDelete".  we can then at least break this up and comment appropriately.  not a big deal since this isn't that hard to grok.  It just gets more and more stuff to think about (IMO) as these expressions get larger and more complex.

Reply:
I'll leave as is for now - but I do agree with the sentiment.